test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_performance_optimized_sm_training.py

```diff
@@ -0,0 +1,194 @@
+@pytest.mark.model("resnet50")
+def test_optimized_tensorflow_sagemaker_training_performance_singlenode(tensorflow_training, region, gpu_only, tf25_and_above_only):
+    throughput_without_xla = run_sm_perf_test(
+                                image_uri=tensorflow_training,
+                                xla=False,
+                                num_nodes=1,
+                                region=region,
+                                threshold=None
+                                )
+    throughput_with_xla = run_sm_perf_test(
+                                image_uri=tensorflow_training,
+                                xla=True,
+                                num_nodes=1,
+                                region=region,
+                                threshold=None
+                                )
+   
+    _, framework_version = get_framework_and_version_from_tag(tensorflow_training)
+    py_version = "py2" if "py2" in tensorflow_training else "py37" if "py37" in tensorflow_training else "py3"
+    '''
+    This is a data collection effort for now.
+    There is enough variation in performance to warrant putting down a THRESHOLD in terms of performance ratio
+    assert throughput_with_xla > throughput_without_xla, (
+        f"tensorflow {framework_version} sagemaker training gpu {py_version} imagenet 1 nodes "
+        f"XLA Benchmark Result {throughput_with_xla} does not reach the threshold {throughput_without_xla}"
+    )
+    '''
+
+
+
+def run_sm_perf_test(image_uri, xla, num_nodes, region, threshold=None):
+    """
+    Run TF sagemaker training performance tests
+
+    Additional context: Setup for this function is performed by 'setup_sm_benchmark_tf_train_env' -- this installs
+    some prerequisite packages, clones some repos, and creates a virtualenv called sm_benchmark_venv.
+
+    TODO: Refactor the above setup function to be more obviously connected to this function,
+    TODO: and install requirements via a requirements.txt file
+
+    :param image_uri: ECR image URI
+    :param xla: [ True | False ] Enable XLA acceleration
+    :param num_nodes: Number of nodes to run on
+    :param region: AWS region
+
+    This function was inspired by deep-learning-containers/test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_performance_tensorflow_sm_training.py
+
+    """
+    _, framework_version = get_framework_and_version_from_tag(image_uri)
+
+    processor = "xla" if xla else "gpu"
+    device_cuda_str = f"{processor}-{get_cuda_version_from_tag(image_uri)}"
+    ec2_instance_type = "p3.8xlarge"
+    py_version = "py2" if "py2" in image_uri else "py37" if "py37" in image_uri else "py3"
+
+    time_str = time.strftime("%Y-%m-%d-%H-%M-%S")
+    commit_info = os.getenv("CODEBUILD_RESOLVED_SOURCE_VERSION")
+    target_upload_location = os.path.join(
+        BENCHMARK_RESULTS_S3_BUCKET, "xla", "tensorflow", framework_version, "sagemaker", "training", device_cuda_str, py_version
```

Comment:
is the additional `xla` folder added to ease in fetching CW metrics?

Reply:
Yes, we want to use a different S3 key since a different set of people will be working working on/utilizing the tests reports here. This design will ensure there will be no overlap between the TF team and XLA team.